sphinx/source/docs/user_guide/bokehjs.rst

```diff
@@ -280,6 +280,93 @@ and hover policy. Here is an example of a ``bar`` chart and the plot it generate
+This makes it so the title cannot be dynamically updated later. In Python, the
+equivalent approach casts the string to the underlying Title() object.
+
+To preserve the ability to update the title later, use the following sequence instead:
+
+.. code-block:: javascript
+
+    var plot = new Bokeh.Plotting.figure()
+    plot.title.text = "some title"
+
+.. _userguide_bokehjs_differences_gridplots:
+
+Using ``sizing_mode`` in gridplot layouts
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+``Bokeh.Plotting.gridplot()`` does not inherit ``sizing_mode`` from the underlying plots.
+To match the expected behavior from the Python equivalent, for example when using
+``sizing_mode: 'stretch_width'``, the ``sizing_mode`` must be explicitly stated for both
+the ``figure()`` and the ``gridplot()`` objects.
+
+.. _userguide_bokehjs_differences_range_padding:
+
+Making axis range properties persitent
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Setting ``plot.x_range.start`` and ``plot.x_range.end``, or ``plot.x_range.range_padding``
+to adjust whitespace around glyphs on an X axis (or the corresponding properties for a
+Y axis) is not persistent across resets. In the Python library, the values for these
+properties are persistent when a figure is reset. To achieve similar behavior in BokehJS,
+set both ``plot.x_range.range_padding`` and ``plot.x_range._initial_range_padding``.
+
+.. _userguide_bokehjs_differences_palette:
+
+Palettes for ``Bokeh.Plotting`` objects
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The color palette is available under ``Bokeh.Charts`` but not under ``Bokeh.Plotting``.
+To use it with ``Bokeh.Plotting`` objects such as ``Bokeh.Plotting.image()``, the
+palette argument must be provided as a color string array, rather than as a string.
+
+.. _userguide_bokehjs_differences_name_field:
+
+Setting glyph names for ``HoverTool``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Calling ``plot.line([0,1],[0,1],{name:“the_name”})`` sets the name under ``renderer.glyph.name``,
+rather than ``renderer.name``. The latter is used for HoverTool, under the special field
+``$name``. To enable this special field in the HoverTool, pass the glyph name to the renderer:
+
+.. code-block:: javascript
+
+    plot.renderers[0].name = plot.renderers[0].glyph.name;
+
+.. _userguide_bokehjs_differences_div_widget:
+
+Using ``Div()`` in a ``gridplot()``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A ``Div()`` instance created by calling ``var mydiv = new Bokeh.Widgets.Div()``
+does not include ``mydiv.toolbar.tools`` to facilitate using in a custom
+``Bokeh.Plotting.gridplot()``; or ``gridplot()`` fails to check whether the
+property exists before accessing. This throws ``Uncaught TypeError: Cannot read
+property 'tools' of undefined``
+
+To use a ``Div()`` inside a ``gridplot()``, make sure to first set the div's
+toolbar property, as follows:
+
+.. code-block:: javascript
+
+    mydiv.toolbar = {tools:[]}
```

Comment:
bokeh's and bokehjs' match at API level, however bokeh's implementation is more permissive, breaking the API promise.

Reply:
It's bad advice to tell users to do this kind of patching, because this taps into bokehjs' internal implementation details, and those can change at any time, breaking users' code.

Reply:
I agree the patch is suboptimal. If there is a better solution, I'd be happy to update these docs edits and use it in my own code! The proposed workaround is what did it for me, though. It being a single-line patch, it is also clean enough that I could tag it for review in my code, should the inner workings change in future Bokeh updates.
It sure would have been nice to find something like it in the docs as I was translating my original python plots to JS, which is why I'm recommending it here (took me some time of hunting to get the JS translation working). If there is no better working solution at the moment, perhaps we can use more direct language to make it clear that this is a temporary workaround?

Reply:
@oviquezr I guess the point is that information lives forever on the internet, even when it goes stale and out of date, and for this reason we should simply refrain from documenting some workaround, if they are will cause more harm than good, in the future. Some things are simply bug to fix, when there are resources to address them, and be left at that.

Reply:
@bryevdv @mattpap  That I can most definitely live with! I'll move the `Div()` bit off this docs update and into its own issue in the tracker. I'm still, slowly, working my way through converting the original list from the discourse topic into hopefully helpful bug reports...